src/django/api/views.py

```diff
@@ -3437,3 +3439,45 @@ def list(self, request):
             FacilityActivityReport.objects.all(), many=True).data
 
         return Response(response_data)
+
+
+class ContributorFacilityListAutoSchema(AutoSchema):
+    def get_serializer_fields(self, path, method):
+        if method == 'GET':
+            return [
+                coreapi.Field(
+                    name='contributors',
+                    location='query',
+                    description=('The contributor ID.'),
+                    required=True,
+                )
+            ]
+        return super(FacilityActivityReportAutoSchema,
+                     self).get_serializer_fields(path, method)
+
+
+@schema(ContributorFacilityListAutoSchema())
+class ContributorFacilityListViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    View active Facility Lists filtered by Contributor.
+    """
+    queryset = FacilityList.objects.filter(source__is_active=True)
+    serializer_class = ContributorListSerializer
+
+    def list(self, request):
+        params = ContributorListQueryParamsSerializer(
+            data=request.query_params)
+
+        if not params.is_valid():
+            raise ValidationError(params.errors)
+
+        contributors = params.data.get('contributors', [])
+
+        response_data = [
+            (list.id, list.name)
+            for list
+            in FacilityList.objects.filter(
+                source__contributor__id__in=contributors).order_by('name')
+        ]
```

Comment:
Updated in 7c1a137. I ended up removing the serializer class entirely, because it wan't necessary or cooperative with the format of the results, but I did switch to using the queryset appropriately.